lib/datadog/core/diagnostics/health.rb

```diff
@@ -30,6 +37,26 @@ class Metrics < Core::Metrics::Client
           gauge :queue_max_length, Ext::Health::Metrics::METRIC_QUEUE_MAX_LENGTH
           gauge :queue_spans, Ext::Health::Metrics::METRIC_QUEUE_SPANS
           gauge :sampling_service_cache_length, Ext::Health::Metrics::METRIC_SAMPLING_SERVICE_CACHE_LENGTH
+
+          def default_metric_options
+            return super unless @tags
+
+            # Return dupes, so that the constant isn't modified,
+            # and defaults are unfrozen for mutation in Statsd.
+            super.tap do |options|
+              options[:tags] = options[:tags].dup
```

Comment:
I noticed we also do the `dup` in `Core::Runtime::Metrics#default_metric_options` but is this really needed?

As far as I see it, `super` here is referring to `Datadog::Core::Metrics::Options#default_metric_options` which already dups the frozen tags so I don't think we need to on any of the child classes...